packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2243,21 +2246,33 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     // current composing region) is very infinite-loop-prone: the formatters
     // will keep trying to modify the composing region while Gboard will keep
     // trying to restore the original composing region.
-    final bool textChanged = _value.text != value.text
-                          || (!_value.composing.isCollapsed && value.composing.isCollapsed);
-    final bool selectionChanged = _value.selection != value.selection;
-
-    if (textChanged) {
-      value = widget.inputFormatters?.fold<TextEditingValue>(
-        value,
-        (TextEditingValue newValue, TextInputFormatter formatter) => formatter.formatEditUpdate(_value, newValue),
-      ) ?? value;
+    final bool preformatTextChanged = _value.text != newTextEditingValue.text
+                                      || (!_value.composing.isCollapsed && newTextEditingValue.composing.isCollapsed);
+
+    final List<TextInputFormatter>? formatters = widget.inputFormatters;
+    if (preformatTextChanged && formatters != null && formatters.isNotEmpty) {
+      for (final TextInputFormatter formatter in formatters) {
+        try {
```

Comment:
So if an error is thrown by a formatter, then that formatter will not be applied, but it will continue on to applying subsequent formatters.  Is that desired or should it die right there?

Reply:
Good point. I wanted to get the type of the formatter but stopping the iteration makes more sense.